### PR TITLE
[MB-2597] Domestic Destination Pricer

### PR DESCRIPTION
**Pull request by garrettqmartin8**
## Description

We need to know how to price domestic destination service items for a shipment, and this PR does just that!

## Setup

- `make db_dev_e2e_populate`
- `make server_run`
- Follow the instructions in the [acceptance testing wiki](https://github.com/transcom/mymove/wiki/Acceptance-Testing-Payment-Requests). **Note:** Make sure when you run the SQL query in order to verify the price, make sure the service area code, e.g. "080" matches the one returned in the payload from creating a payment request through the prime API client.

For reference: the price I got back from the payload was 9052 cents, and the number I got from manually verifying the price calculation was 9,051.7014.

## Code Review Verification Steps

* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-2597) for this change

### CODE REVIEW

File: migrations/app/migrations_manifest.txt [hunk 1]
```diff
@@ -524,3 +524,4 @@
 20200624145812_create_ghc_diesel_fuel_prices.up.sql
 20200624211813_add_contract_to_service_domestic_linehaul.up.sql
 20200701133913_add_days_in_storage_and_requested_delivery_date.up.sql
+20200715153454_add_contract_code_param_to_ddp.up.sql
```
[the remark this answers is not in the document]
Comment (garrettqmartin8) [reply migrations/app/migrations_manifest.txt:527]: This was generated by running `milmove gen migration -n migration_name`.

File: migrations/app/schema/20200715153454_add_contract_code_param_to_ddp.up.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO service_params(id, service_id, service_item_param_key_id, created_at, updated_at)
+VALUES ('c5e46220-b1ad-4388-b8b0-1128c92f0bc0', (SELECT id from re_services WHERE code = 'DDP'), (SELECT id FROM service_item_param_keys where key = 'ContractCode'), now(), now())
```
[the remark this answers is not in the document]
Comment (garrettqmartin8) [reply migrations/app/schema/20200715153454_add_contract_code_param_to_ddp.up.sql:2]: This is a new id that I created by running `uuidgen | tr \"[:upper:]\" \"[:lower:]\"` on the command line.

File: pkg/services/ghcrateengine/domestic_destination_pricer.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	if weight < minDomesticWeight {
+		return 0, fmt.Errorf("Weight must be a minimum of %d", minDomesticWeight)
+	}
+	if len(serviceArea) == 0 {
+		return 0, errors.New("ServiceArea is required")
+	}
+
+	isPeakPeriod := IsPeakPeriod(requestedPickupDate)
+	// look up rate for domestic destination price
+	shorthaulServiceCode := "DDP"
```
Comment (jacquelineIO) [inline pkg/services/ghcrateengine/domestic_destination_pricer.go:45]: Why is this hardcoded?

Comment (garrettqmartin8) [reply pkg/services/ghcrateengine/domestic_destination_pricer.go]: Changed this to use a string constant 👍

**Pull request merged**